pkg/duckdb/ephemeral_db.go

```diff
@@ -5,58 +5,659 @@ package duck
+			}
+			switch v := value.(type) {
+			case float64:
+				switch d := dest[i].(type) {
+				case *float32:
+					*d = float32(v)
+				case *float64:
+					*d = v
+				}
+			default:
+				return fmt.Errorf("cannot convert %T to float at column %d", value, i)
+			}
+		case *bool:
+			if value == nil {
+				*d = false
+			} else if b, ok := value.(bool); ok {
+				*d = b
+			} else {
+				return fmt.Errorf("cannot convert %T to bool at column %d", value, i)
+			}
+		case sql.Scanner:
+			if err := d.Scan(value); err != nil {
+				return fmt.Errorf("error scanning column %d: %w", i, err)
+			}
+		default:
+			return fmt.Errorf("unsupported destination type for column %d: %T", i, dest[i])
 		}
-	}(conn)
+	}
 
-	return conn.QueryRowContext(ctx, query, args...)
+	return nil
+}
+
+// adbcColumnType implements the ColumnType interface.
+type adbcColumnType struct {
+	name   string
+	dbType string
+}
+
+func (c *adbcColumnType) DatabaseTypeName() string {
+	return c.dbType
+}
+
+// getArrowValue extracts a value from an Arrow array at the given index.
+func getArrowValue(arr arrow.Array, idx int) interface{} {
+	if arr.IsNull(idx) {
+		return nil
+	}
+
+	switch arr := arr.(type) {
+	case *array.Int8:
+		return int64(arr.Value(idx))
+	case *array.Int16:
+		return int64(arr.Value(idx))
+	case *array.Int32:
+		return int64(arr.Value(idx))
+	case *array.Int64:
+		return arr.Value(idx)
+	case *array.Uint8:
+		return uint64(arr.Value(idx))
+	case *array.Uint16:
+		return uint64(arr.Value(idx))
+	case *array.Uint32:
+		return uint64(arr.Value(idx))
+	case *array.Uint64:
+		return arr.Value(idx)
+	case *array.Float32:
+		return float64(arr.Value(idx))
+	case *array.Float64:
+		return arr.Value(idx)
+	case *array.String:
+		// Explicitly copy the string to ensure it's not a view into Arrow memory
+		s := arr.Value(idx)
+		// Force a copy by converting to []byte and back
+		return string(append([]byte(nil), []byte(s)...))
+	case *array.LargeString:
+		// Explicitly copy the string to ensure it's not a view into Arrow memory
+		s := arr.Value(idx)
+		// Force a copy by converting to []byte and back
+		return string(append([]byte(nil), []byte(s)...))
+	case *array.Binary:
+		return string(arr.Value(idx))
+	case *array.LargeBinary:
+		return string(arr.Value(idx))
+	case *array.FixedSizeBinary:
+		return string(arr.Value(idx))
+	case *array.Boolean:
+		return arr.Value(idx)
+	case *array.Date32:
+		return arr.Value(idx).ToTime()
+	case *array.Date64:
+		return arr.Value(idx).ToTime()
+	case *array.Timestamp:
+		return arr.Value(idx).ToTime(arr.DataType().(*arrow.TimestampType).Unit)
+	case *array.Time32:
+		return arr.Value(idx)
+	case *array.Time64:
+		return arr.Value(idx)
+	case *array.Decimal128:
+		val := arr.Value(idx)
+		dt := arr.DataType().(*arrow.Decimal128Type)
+		f := val.ToFloat64(dt.Scale)
+		// Format with exact precision to avoid floating-point representation issues
+		// For scale 2, format as "%.2f"
+		formatted := fmt.Sprintf("%.*f", dt.Scale, f)
```

Comment:
Could you use the decimals ToString function for this? 

Should this be something we add to the actual decimal type?